nixos/modules/config/qt.nix

```diff
@@ -0,0 +1,249 @@
+      {
+        font = cfg.font.name;
+        menuFont = cfg.font.name;
+        toolBarFont = cfg.font.name;
+      }
+    //
+    optionalAttrs (cfg.theme != null)
+      { widgetStyle = cfg.theme.name; };
+  icons =
+    optionalAttrs (cfg.iconTheme != null)
+      { Theme = cfg.iconTheme.name; };
+
+  qt =
+    optionalAttrs (cfg.font != null)
+      { font = ''"${cfg.font.name}"''; }
+    //
+    optionalAttrs (cfg.theme != null)
+      { style = "GTK+"; };
+
+  fontType = types.submodule {
+    options = {
+      package = mkOption {
+        internal = true;
+        type = types.nullOr types.package;
+        default = null;
+      };
+      name = mkOption {
+        internal = true;
+        type = types.str;
+      };
+    };
+  };
+  themeType = types.submodule {
+    options = {
+      package = mkOption {
+        internal = true;
+        type = types.nullOr types.package;
+        default = null;
+      };
+      name = mkOption {
+        internal = true;
+        type = types.str;
+      };
+    };
+  };
+
+  optionalPackage = opt:
+    optional (opt != null && opt.package != null) opt.package;
+
+  platforms = {
+    gtk2 = rec {
+      description = ''
+        <varlistentry>
+          <term><literal>gtk2</literal></term>
+          <listitem><para>Use GTK2 theme with
+            <link xlink:href="https://github.com/qt/qtstyleplugins">qtstyleplugins</link>
+          </para></listitem>
+        </varlistentry>
+      '';
+      themes = [ "cleanlooks" "gtk2" "cde" "motif" "plastique" ];
+
+      assertions = [
+        {
+          assertion = cfg.theme != null && any (name: name == cfg.theme.name) themes;
+          message = "`qt.theme.name` is not one of [ ${toString themes} ].";
+        }
+        {
+          assertion = cfg.font == null && cfg.iconTheme == null;
+          message = "`qt.font` and `qt.iconTheme` are only supported by kde platform.";
+        }
+      ];
+      environment.variables.QT_QPA_PLATFORMTHEME = "gtk2";
+      environment.variables.QT_STYLE_OVERRIDE = cfg.theme.name;
+      environment.systemPackages = [ pkgs.libsForQt5.qtstyleplugins ];
+    };
+    qgnomeplatform = {
+      description = ''
+        <varlistentry>
+          <term><literal>qgnomeplatform</literal></term>
+          <listitem><para>Use GNOME theme with
+            <link xlink:href="https://github.com/FedoraQt/QGnomePlatform">qgnomeplatform</link>
+          </para></listitem>
+        </varlistentry>
+      '';
+
+      assertions = [
+        {
+          assertion = cfg.font == null && cfg.iconTheme == null;
+          message = "`qt.font` and `qt.iconTheme` are only supported by kde platform.";
+        }
+      ];
+      environment.variables.QT_QPA_PLATFORMTHEME = "qgnomeplatform";
+      # TODO: make this optional once https://github.com/NixOS/nixpkgs/issues/54150 is fixed
+      # qgnomeplatform reads theme and other settings from dconf db
+      environment.variables.QT_STYLE_OVERRIDE = cfg.theme.name;
```

Comment:
@worldofpeace In the past it was limited to `adwaita`, but actually `qgnomeplatform` allows any theme to use

Reply:
Are we sure, when I tested it by infinisil's request I got segfaults...

Reply:
@worldofpeace It works for me with other themes like breeze